lib/sample_uploader/sample_uploaderImpl.py

```diff
@@ -767,6 +767,36 @@ def filter_samplesets(self, ctx, params):
                              'output is not type dict as required.')
         # return the results
         return [output]
+
+    def get_sampleset_meta(self, ctx, params):
+        """
+        :param params: instance of type "GetSamplesetMetaParams" (Get list of
+           metadata keys/columns from a given sampleset. Used to populate
+           filter_sampleset dynamic dropdown with valid options from a given
+           list of samples.) -> structure: parameter "sampleset_ref" of list
+           of String
+        :returns: instance of list of String
+        """
+        # ctx is the context object
+        # return variables are: results
+        #BEGIN get_sampleset_meta
+        samples = []
+        for sample_set in self.dfu.get_objects({'object_refs': params['sample_set_ref']})['data']:
+            samples.extend(sample_set['data']['samples'])
+        sample_ids = [{'id': sample['id'], 'version':sample['version']} for sample in samples]
```

Comment:
Should be probably fine. But I would call `sample.get('id') and sample.get('version')` here. Or better do a check here before calling the search API. Because `sample_set_ref` can theoretically be any object type. If objects other than `SampleSet` were given, this probably will raise a KeyError.